api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -135,4 +168,170 @@ public void validateInstitutionalAffiliation(Profile profile) {
+        demographicSurveyMapper.demographicSurveyToDbDemographicSurvey(
+            updatedProfile.getDemographicSurvey());
+
+    if (user.getDemographicSurveyCompletionTime() == null && dbDemographicSurvey != null) {
+      user.setDemographicSurveyCompletionTime(now);
+    }
+
+    if (dbDemographicSurvey != null && dbDemographicSurvey.getUser() == null) {
+      dbDemographicSurvey.setUser(user);
+    }
+
+    user.setDemographicSurvey(dbDemographicSurvey);
+    user.setLastModifiedTime(now);
+
+    updateInstitutionalAffiliations(updatedProfile, user);
+    boolean requireInstitutionalVerification =
+        workbenchConfigProvider.get().featureFlags.requireInstitutionalVerification;
+    if (requireInstitutionalVerification) {
+      validateInstitutionalAffiliation(updatedProfile);
+    }
+
+    userService.updateUserWithConflictHandling(user);
+    if (requireInstitutionalVerification) {
+      DbVerifiedInstitutionalAffiliation updatedDbVerifiedAffiliation =
+          verifiedInstitutionalAffiliationMapper.modelToDbWithoutUser(
+              updatedProfile.getVerifiedInstitutionalAffiliation(), institutionService);
+      updatedDbVerifiedAffiliation.setUser(user);
+      Optional<DbVerifiedInstitutionalAffiliation> dbVerifiedAffiliation =
+          verifiedInstitutionalAffiliationDao.findFirstByUser(user);
+      dbVerifiedAffiliation.ifPresent(
+          verifiedInstitutionalAffiliation ->
+              updatedDbVerifiedAffiliation.setVerifiedInstitutionalAffiliationId(
+                  verifiedInstitutionalAffiliation.getVerifiedInstitutionalAffiliationId()));
+      this.verifiedInstitutionalAffiliationDao.save(updatedDbVerifiedAffiliation);
+    }
+
+    final Profile appliedUpdatedProfile = getProfile(user);
+    profileAuditor.fireUpdateAction(previousProfile, appliedUpdatedProfile);
+  }
+
+  public void validateAndCleanProfile(Profile profile) throws BadRequestException {
+    // Validation steps, which yield a BadRequestException if errors are found.
+    String userName = profile.getUsername();
+    if (userName == null || userName.length() < 3 || userName.length() > 64) {
+      throw new BadRequestException(
+          "Username should be at least 3 characters and not more than 64 characters");
+    }
+    validateStringLength(profile.getGivenName(), "Given Name", 80, 1);
+    validateStringLength(profile.getFamilyName(), "Family Name", 80, 1);
+
+    // Cleaning steps, which provide non-null fields or apply some cleanup / transformation.
+    profile.setDemographicSurvey(
+        Optional.ofNullable(profile.getDemographicSurvey()).orElse(new DemographicSurvey()));
+    profile.setInstitutionalAffiliations(
+        Optional.ofNullable(profile.getInstitutionalAffiliations()).orElse(new ArrayList<>()));
+    // We always store the username as all lowercase.
+    profile.setUsername(profile.getUsername().toLowerCase());
+  }
+
+  private void updateInstitutionalAffiliations(Profile updatedProfile, DbUser user) {
```

Comment:
delete ProfileController version of this
move ProfileController comment and @Deprecated here